src/file-write.ts

```diff
@@ -0,0 +1,135 @@
+import {
+    fsReadPromise,
+    fsRenamePromise,
+    fsUnlinkPromise,
+    fsWritePromise,
+    getNextBufferSubarrayAsync,
+    getNextBufferSubarraySync,
+    processFile,
+    processFileAsync
+} from "./util-file"
+import * as tmp from 'tmp'
+import * as path from 'path'
+import * as fs from 'fs'
+import { findId3TagPosition, getId3TagSize } from "./id3-tag"
+
+const FileBufferSize = 20 * 1024 * 1024
+
+export function writeId3TagToFileSync(filepath: string, id3Tag: Buffer) {
+    const tmpFile = getTmpFilePathSync(filepath)
+    processFile(filepath, 'r', (readFileDescriptor) => {
+        processFile(tmpFile, 'w', (writeFileDescriptor) => {
+            fs.writeSync(writeFileDescriptor, id3Tag)
+            streamOriginalIntoNewFileSync(readFileDescriptor, writeFileDescriptor)
+        })
+    })
+    fs.unlinkSync(filepath)
+    fs.renameSync(tmpFile, filepath)
+}
+
+export function writeId3TagToFileAsync(filepath: string, id3Tag: Buffer, callback: (err: Error|null) => void) {
```

Comment:
Ideally we could optimize by doing the process of the buffer during an async read/write operation, but this is a bit more complicated, let's leave it for another PR.